src/precice/bindings/python_future/precice_future.pyx

```diff
@@ -118,7 +118,7 @@ cdef class Interface:
         if not isinstance(position, np.ndarray):
             position = np.asarray(position)
         dimensions = position.size
-        assert(dimensions == self.get_dimensions())
+        assert(dimensions == self.get_dimensions(), "Function set_mesh_vertex expects a 1d numpy array with coordinates of point as position input")
```

Comment:
I think it would generally be helpful to provide error messages where you give the user information about what was received and what the interface accepts.

Example:

```
assert(dimensions == self.get_dimensions(), "Function set_mesh_vertex(mesh_id, position) received position of type {position_type}, but expects a 1d numpy array with coordinates of point as position input".format(type(position)))
```

For other API calls, we might need to provide additional information, like expected shape etc.

Reply:
It might even be a good idea to refactor these checks out into some new functions to avoid code duplication.

Reply:
Is there a chance that the `type()` call to a wrongly defined variable or array lead to further more confusing error output ? That was my thinking towards restraining to pass additional arguments to the error message.

Reply:
It can always become worse, I assume (especially, in python, where you can break everything, if you really want it). But for the normal use case that a user provides a list or another iterable object, while we expect a numpy array, it should work.

Reply:
I like the new assertions much more than the old ones. :)

From my perspective it would be important 

- to know that something failed
- to know why it failed or at least to get a hint that helps me figuring this out. 
- to maybe know where it failed (if I need to know that/helps me solve the problem.) 

The new assertions seem to address these points as far I as I checked.